spyder_kernels/comms/commbase.py

```diff
@@ -0,0 +1,407 @@
+        Parameters
+        ----------
+        message_id : str
+            The identifier for the message
+        handler : callback
+            A function to handle the message. This is called with 3 arguments:
+                - msg_dict: A dictionary with message information.
+                - buffer: The data transmitted in the buffer
+                - load_exception: Exception from buffer unpickling
+            Pass None to unregister the message_id
+        """
+        if handler is None:
+            self._message_handlers.pop(message_id, None)
+            return
+
+        self._message_handlers[message_id] = handler
+
+    def _register_comm(self, comm):
+        """
+        Open a new comm to the kernel.
+        """
+        comm.on_msg(self._comm_message)
+        comm.on_close(self._comm_close)
+        self._comms[comm.comm_id] = comm
+
+    def _comm_close(self, msg):
+        """Close comm."""
+        comm_id = msg['content']['comm_id']
+        del self._comms[comm_id]
+
+    def _comm_message(self, msg):
+        """
+        Handle internal spyder messages.
+        """
+        self.calling_comm_id = msg['content']['comm_id']
+        # Load the buffer. Only one is supported.
+        try:
+            if PY2:
+                buffer = cloudpickle.loads(msg['buffers'][0])
+            else:
+                buffer = cloudpickle.loads(bytes(msg['buffers'][0]))
+            load_exception = None
+        except Exception as e:
+            load_exception = e
+            buffer = None
+
+        # Get message dict
+        msg_dict = msg['content']['data']
+
+        spyder_msg_type = msg_dict['spyder_msg_type']
+
+        if spyder_msg_type in self._message_handlers:
+            self._message_handlers[spyder_msg_type](
+                msg_dict, buffer, load_exception)
+            return
+
+        logger.debug("No such spyder message type: %s" % spyder_msg_type)
+
+    def _handle_remote_call(self, msg, buffer, load_exception):
+        """Handle a remote call."""
+        msg_dict = msg['content']
+        if load_exception:
+            logger.debug(
+                "Exception in cloudpickle.loads : %s" % str(load_exception))
+            return
+        try:
+            return_value = self._remote_callback(
+                    msg_dict['call_name'],
+                    buffer['call_args'],
+                    buffer['call_kwargs'])
+            self._set_call_return_value(msg_dict, return_value)
+        except Exception as e:
+            tb = traceback.extract_tb(sys.exc_info()[2])
+            self._set_call_return_value(msg_dict, e, traceback=tb)
+
+    def _remote_callback(self, call_name, call_args, call_kwargs):
+        """Call the callback function for the remote call."""
+        if call_name in self._remote_call_handlers:
+            return self._remote_call_handlers[call_name](
+                *call_args, **call_kwargs)
+
+        raise CommError("No such spyder call type: %s" % call_name)
```

Comment:
Let's change this to

```python
else:
    logger.debug("No such spyder call type: %s" % call_name)
```

as above.

Reply:
I want to notify the other side and the exception is catched above.

Reply:
Ok, fair enough.